### PR TITLE
WIP:ISPN-14008 Docs indexing annotations

### PR DESCRIPTION
**Pull request by dvagnero**
https://issues.redhat.com/browse/ISPN-14008
Infinispan indexing annotations
needs backport to 14

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/code_examples/AnalyzerDefinitions.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
-@Field(store = Store.YES, analyze = Analyze.YES, analyzer = @Analyzer(definition = "keyword"))
```
Comment (dvagnero) [inline documentation/src/main/asciidoc/topics/code_examples/AnalyzerDefinitions.java:1]: @fax4ever could you please review the examples?

**Review by fax4ever — COMMENTED** [review]
@dvagnero I reviewed the examples

File: documentation/src/main/asciidoc/topics/con_indexing_annotations.adoc [hunk 2]
```diff
@@ -1,52 +1,75 @@
...
+Supports `searchable` and `projectable` attributes.
+
+@Embedded:: Use this annotation for embedded indexes and specify the entity structure.
+The `NESTED` structure preserves the original object relationship structure while the `FLATTENED` structure makes the leaf fields multivalued of the parent entity.
+
+//TODO update the table with attributes
 [%autowidth,cols="1,1,1",stripes=even]
 |===
-|Attribute |Description |Values
+|Attribute |Description |Values | Annotation
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply documentation/src/main/asciidoc/topics/con_indexing_annotations.adoc:32]: |            | searchable | sortable | projectable | aggregable | indexNullAs | decimalScale | normalizer | norms | analyzer | searchAnalyzer | termVector |
|------------|------------|----------|-------------|------------|-------------|--------------|------------|-------|----------|----------------|------------|
| `@Basic`   | yes        | yes      | yes         | yes        | yes         | no           | no         | no    | no       | no             | no         |
| `@Decimal` | yes        | yes      | yes         | yes        | yes         | yes          | no         | no    | no       | no             | no         |
| `@Keyword` | yes        | yes      | yes         | yes        | yes         | no           | yes        | yes   | no       | no             | no         |
| `@Text`    | yes        | no       | yes         | no         | no          | no           | no         | yes   | yes      | yes            | yes        |

**Review by fax4ever — COMMENTED** [review]
Thank you @dvagnero I reviewed the rest.
Nice job, only a comment here.

In the previous review https://github.com/infinispan/infinispan/pull/10785#pullrequestreview-1372823376, you can find the review of the examples.

File: documentation/src/main/asciidoc/topics/con_indexing_annotations.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvagnero) [reply documentation/src/main/asciidoc/topics/con_indexing_annotations.adoc]: Thank you @fax4ever I updated the table and descriptions of annotations. I'll check the procedures and examples tomorrow. 
Do we need some description for attributes?

**Review by fax4ever — CHANGES_REQUESTED** [review]
I think that the suggestions are not properly applied

**Review by fax4ever — APPROVED** [review]

**Comment by fax4ever** [comment]
Thank you @dvagnero. Great job here!
Going to merge it.

**Pull request merged**